lib/index.js

```diff
@@ -191,12 +191,16 @@ CloudFront.prototype.request = function(method, path, headers, cb) {
 
       cb(err, info);
     });
-    parser.on("error", function(err) {
-      cb(err, info);
+    parser.on('error', function(err) {
+      cb(err);
     });
 
   });
 
+  req.on('error', function(err) {
```

Comment:
I mean this code will be executed before the response event is received anyway so it would work even if the error event is received after the response event is received.